internal/git/gitsync_processor.go

```diff
@@ -106,7 +106,7 @@ func watchRepo(ctx context.Context, r *git.Repository, gitSync *v1.GitSync, rest
 		logger.Errorw("error resolving the revision", "revision", repo.TargetRevision, "err", err, "repo", repo.RepoUrl)
 		return err
 	}
-
+	lastCommitHash = hash.String()
```

Comment:
I think we need to set `lastCommitHash` after this line below:
```
val, ok := gitSync.Status.CommitStatus[repo.Name]
```
We should add for the `ok` case:
```
lastCommitHash = gitSync.Status.CommitStatus[repo.Name].Hash
```

Reply:
This is the case where the Controller just restarted, and we get the last hash from the CommitStatus entry

Reply:
And what if we donot have any entry ? or is it the very first time cloning an syncing @juliev0

Reply:
I'm also thinking it logically makes sense for readability to move this line to the `!ok` case after we apply the resources because that's where the `lastCommitHash` truly is the last one set.

Reply:
> And what if we donot have any entry ? or is it the very first time cloning an syncing @juliev0

that's the `!ok` case I mention above